File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int count=0,lcount=0,rcount=0,i=0;
        while(i<s.length()){
            if(s.charAt(i)=='L'){
                lcount++;
            }else{
                rcount++;
            }
            if(lcount==rcount){
                count++;
                lcount=0;
                rcount=0;
            }
            i++;
        }
        return count;
    }
}

File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        countl=countr=count=0
        for i in s:
            if i=='L':
                countl+=1
            else:
                countr+=1
            if countl==countr:
                count+=1
                countl= countr=0
               
        return count        
        